src/compiler/factory.ts

```diff
@@ -1260,11 +1268,11 @@ namespace ts {
                     foundUseStrict = true;
                 }
 
-                target.push(source[i]);
+                target.unshift(source[i]);

```

Comment:
But prologues need to go at the beginning, so what situations will you have where you're adding prologues other than to the beginning of the array? Any time this function gets called after anything, that's probably a bad sign.

> But there are the same and it will not matter right?

Apart from preserving what the user wrote the best we can, that's unfortunately not necessarily true. Check out

> Implementations may define implementation specific meanings for ExpressionStatement productions which are not a Use Strict Directive and which occur in a Directive Prologue.

So you don't actually know what your engine will want to do for some prologues and whether they are order-dependent. For instance, there used to be uncertainty on whether the `"use asm` directive should be allowed to come before a `"use strict"`. https://bugzilla.mozilla.org/show_bug.cgi?id=877908